src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -615,7 +582,8 @@ private void EndRequest(HttpContext httpContext, long timestamp)
 
                 this.client.TrackRequest(telemetry);
 
-                var activity = httpContext?.Features.Get<Activity>();
+                // Stop what we started.
+                var activity = Activity.Current; 
```

Comment:
This was an unrelated fix. Activity was never stored into httpContext.Features, so we never stopped the activity we started. Good code hygiene.

Reply:
Could it be that framework store it for us? I do not know, so I would've guessed it may.

Reply:
Framework (hosting) don't store it. SDK creates and starts activity with the name "ActivityCreatedByHostingDiagnosticListener", and framework has no knowledge of it.